File: be/src/main/java/com/example/be/common/domain/exam/entity/Answer.java
package com.example.be.common.domain.exam.entity;

import jakarta.persistence.*;
import lombok.*;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "answers")
public class Answer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String answerText;

    @Lob
    @Column(columnDefinition = "TEXT")
    private String explanation;

    @OneToOne
    @JoinColumn(name = "question_id")
    private Question question;
}

File: gpt_be/models/answer.py
from sqlalchemy import Column, Integer, String, ForeignKey, Text
from sqlalchemy.orm import relationship
from database import Base

class Answer(Base):
    __tablename__ = "answers"

    id = Column(Integer, primary_key=True, index=True)
    answer_text = Column(String)
    explanation = Column(Text)
    question_id = Column(Integer, ForeignKey("questions.id"))

    # 관계 설정
    question = relationship("Question", back_populates="answer") 